sway-core/src/transform/to_parsed_lang/convert_parse_tree.rs

```diff
@@ -2496,19 +2496,40 @@ fn configurable_field_to_constant_declaration(
     attributes: AttributesMap,
 ) -> Result<ParsedDeclId<ConstantDeclaration>, ErrorEmitted> {
     let span = configurable_field.name.span();
+
+    let type_ascription = ty_to_type_argument(context, handler, engines, configurable_field.ty)?;
+
+    let value = expr_to_expression(context, handler, engines, configurable_field.initializer)?;
+    let value = if context.experimental.new_encoding {
+        let call_encode =
+            ExpressionKind::FunctionApplication(Box::new(FunctionApplicationExpression {
+                call_path_binding: TypeBinding {
+                    inner: CallPath {
+                        prefixes: vec![],
```

Comment:
The idea is the same as in other places. We want to enable custom implementation of this. Until we have `#[abi_encode]` attribute, this is the best we can do.

If we enforce the full path now, we will be demanding access to `core`. Something that we are trying to avoid when possible.

Reply:
Ok. I'll stop asking this question in the future I promise :smile: But my bewilderment with this design choice remains **huge**. Dependency on `core` we anyhow have in `contract`s, `script`s, and `predicate`s unless someone wants to write its own `entry` and _everything_ around encoding. We do not know if such need or wish will ever exist, but still opt to render every function named `encode` as invalid at this point, essentially `encode` being a reserved word for function names (`abi_decode` as well, but being not so common word...).

What would cost us to insist on `core::codec` until _eventually_ someone actually really asks for custom implementation and then we either implement and document `#[abi_encode]` or just temporary remove prefixes?

Enough ranting, no need to answer on this one :smile: